solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -0,0 +1,120 @@
+"""
+Make benchmark irradiance and power forecasts.
+"""
+from inspect import signature
+
+from solarforecastarbiter import datamodel, pvmodel
+
+
+def run(site, model):
+    """
+    Calculate benchmark irradiance and power forecasts for a site.
+
+    The meaning of the timestamps (instantaneous or interval average)
+    is determined by the model processing function.
+
+    Parameters
+    ----------
+    site : datamodel.Site
+
+    Returns
+    -------
+    ghi : pd.Series
+    dni : pd.Series
+    dhi : pd.Series
+    temp_air : None or pd.Series
+    wind_speed : None or pd.Series
+    ac_power : None or pd.Series
+
+    Examples
+    --------
+    The following code would return hourly average forecasts derived
+    from the subhourly HRRR model.
+
+    >>> from solarforecastarbiter import datamodel
+    >>> from solarforecastarbiter.reference_forecasts import models
+    >>> modeling_parameters = datamodel.FixedTiltModelingParameters(
+    ...     ac_capacity=10, dc_capacity=15,
+    ...     temperature_coefficient=-0.004, dc_loss_factor=0,
+    ...     ac_loss_factor=0)
+    >>> power_plant = datamodel.SolarPowerPlant(
+    ...     name='Test plant', latitude=32.2, longitude=-110.9,
+    ...     elevation=715, timezone='America/Phoenix',
+    ...     modeling_parameters = modeling_parameters)
+    >>> ghi, dni, dhi, temp_air, wind_speed, ac_power = run(
+    ...     power_plant, hrrr_subhourly_to_hourly_mean)
+    """
+
+    *solpos_forecast, resampler = model(
+        site.latitude, site.longitude, site.elevation)
+
+    if isinstance(site, datamodel.SolarPowerPlant):
+        solpos_forecast = maybe_calc_solar_position(site, *solpos_forecast)
+        ac_power = pvmodel.irradiance_to_power(site.modeling_parameters,
+                                               *solpos_forecast)
+    else:
+        ac_power = None
+
+    # resample data after power calculation
+    ret = list(map(resampler, (*solpos_forecast[2:], ac_power)))
+    return ret
+
+
+def maybe_calc_solar_position(site, apparent_zenith, azimuth, *args):
```

Comment:
I strongly dislike ``maybe_`` functions when debugging

Reply:
are you opposed to the uncertainty about if solar position needs to be calculated or are you opposed to the fact that it's wrapped up in a function instead of inlined in the calling block?

Reply:
first mostly, although it is worse that the variable is reassigned after maybe_